### ISSUE TITLE
Lost Trinkets too powerful, causes several mods to be less useful

### ISSUE THREAD

**Issue by dgobi** [NillerMedDild/Enigmatica6#792]
**Modpack version**
Enigmatica 6 0.3.4

**Issue**
Lost Trinkets is a great mod, but it is so powerful as to invalidate a lot of utility from several mods. For example, within hours of starting on a private server, myself and several friends had unlocked creative flight, step assist, no fall damage, and ultra speed. This means I have little incentive for a Sojourner's Sash, Jetpack, etc, which to me is unfortunate.

For balancing I feel the powerful effects should be removed, scaled back, or perhaps gated in some way. I understand this is a subjective observation so of course feel free to close if you disagree. :) 

Thank you for this and all the Enigmatica packs. I love them dearly and appreciate all your hard work.

**Comment by Octr** [comment]
It is a decent mod, but highly unbalanced and easy to abuse in it's current state.

It offers too much without any form of progresssion.

**Comment by MuteTiefling** [comment]
Yeah, there are some of these that are just too good, and they're ultimately easy to farm.

Propose disabling the following as they're available from multiple other more balanced sources:
horseshoe: Step assist 
Rock candy: super speed
Tha Cloud: Fall damage protection
Turtle Shell: Water Breathing
Blaze Heart: Full fire/lava immunity
Tha Bat: Night Vision
Tea Leaf: Wither/poison immunity. Probably too good. Maybe ok
Dark Egg: Pretty sure this summons a bunch of Vex to help you. But they don't seem to go away... 
octopick: Vein miner
Magic Herbs: Immunity to all negative effects
Magical Feathers: Creative Flight
Mad Aura: Immunity to arrows


I know this seems like a long list, but that still leaves a fair number of very unique and situational trinkets.

**Comment by MaxFirestorm** [comment]
*Hard* disagree on disabling them. I've been playing for a while and only just got some of the "good" ones *after* I would have required (or needed) them (for example the Blaze Heart after I've finished setting up a Blaze Farm), thereby making future stuff easier, but I've already put the work into where I would need them

I'm all for gating them or making them more difficult to obtain but *please* don't disable them =(

**Comment by Octr** [comment]
> _Hard_ disagree on disabling them. I've been playing for a while and only just got some of the "good" ones _after_ I would have required (or needed) them (for example the Blaze Heart after I've finished setting up a Blaze Farm), thereby making future stuff easier, but I've already put the work into where I would need them
> 
> I'm all for gating them or making them more difficult to obtain but _please_ don't disable them =(

Some just are unnecessary and ruin the game progression such as creative flight, there are many balanced alternatives that are made useless by trinkets.

Also in it's current state they are entirely rng, just because you got yours in a seemingly progressive way, its highly unlikely to happen that way for most players.

All of the trinkets can be auto farmed early game and you can become godlike before even killing a wither/ender dragon and progressing even a fraction through mods.

The only reason some should be disabled is because there are already alternatives causing redundant overlap

**Comment by Trostmm** [comment]
Any way to limit the number of slots and/or of a certain 'type'. 
IE - You could have all the common ones enables, but only 1 legendary, 1 epic etc?

I like the situational aspect, but I agree having all those effects w/out any ongoing cost beyond the XP to unlock seems unrealistic.

**Comment by RixMixed** [comment]
I am ignorant to how the mod and configs work but I think making the redundant/op trinkets acquirable through very expensive recipes or something similar as an endgame goal could be neat.

**Comment by lamb3664** [comment]
From my experience, you just seem to just get them whenever; they aren't physical items and they aren't crafted.

**Comment by stefment** [comment]
I will miss rock candy. It seems to be the way to go get useful horizontal speed with a jetpack.

**Comment by NillerMedDild** [comment]
> Horseshoe: Step assist

I disagree on this one, as it's a very minor effect. I have disabled all the others in that list. Sorry for those who will miss them, but I have to agree that they are simply too easy to get for their effects, and actually take away from the pack by removing incentive to progress in mods.

**Comment by kil0byt3** [comment]
Love this change, these are very OP. Thanks for the great balance here. 

> Edit: We have no way of gating them sadly.

Just a note here, the trinkets can be unlocked through obtaining the JEI item, and right-clicking with it in hand, so you could turn off the random unlocks from killing/farming/mining etc, and just make some mid-lategame recipes to obtain the item, instead of the random unlocks, but that's a lot more work and a complete change of approach, so I'll leave the decision in your hands :D

**Comment by stefment** [comment]
People who are new to the pack dont know about these trinkets anyway. Its just a nice surprise for them slowly getting these benefits. But people who know what to do can "easily" exploit it and get all the trinkets fast. I feel like new people were just robbed of an experience because more experienced players know how to "exploit". But thats just my humble opinion.

**Comment by ChaosColors** [comment]
I agree with stefment, i just started and loved getting random trinkets, and Tha Cloud is my favorite one
I think you're over nerfing a fun mod for "Balance"

**Comment by MuteTiefling** [comment]
The thing is, everything that's being disabled can still be obtained through other mods. 

Even players who are new to the pack have gotten 'lucky' and gotten things like flight and fire immunity day one... and they've largely agreed it's too much. It kinda saps your will to go explore some of the more interesting methods available to achieve these things.

Think of it like this. What if you started playing and on your first day, you got a creative dispenser upgrade for a drawer. You'd have exactly no reason to go out and explore mods. Need more iron? Yeah, you could work on automating bees or nature's aura. Or work on getting a new pick to try out. Or you could just slap that thing in a drawer and extract all the iron you'll ever need. Rinse, repeat for every material. 

This might seem fun at first, and you might even have fun for a few minutes... but it quickly loses it's charm and ruins every aspect of the game for you. Then you drop the pack and move on. We'd like people to have a longer, more engaging play experience. 

That's what a lot of these trinkets represent. Creative flight first night? Combat is over. No need to worry about any other flight mechanism, of which there are many. Immunity to negative effects? Oof. That's a lot of challenge gone and no need to look for alternative means of protecting yourself.

What remains are actually all rather unique in what they do, which makes them rather more exciting to get. You will not find most of these effects through any other means. I personally think that makes for a better experience for the new player: discovering something truly new, rather than getting handouts.

**Comment by TopPercentageRattata** [comment]
> 
> 
> The thing is, everything that's being disabled can still be obtained through other mods.
The rock candy speed effect is sort of unique in how it applies forward motion
And dark egg vexs despawn just fine once their timer runs out similar to how vanilla vexes do.

**Comment by NillerMedDild** [comment]
>  And dark egg vexs despawn just fine once their timer runs out similar to how vanilla vexes do.

Will re-enable that one then, thanks.

**Comment by HalJordan** [comment]
I disagree on disabling it alltogether. If you don't like how op it is ... don't use it. It's one thing to tweak the configs a bit to nerf it a bit, but nobody is bending your arm behind your back forcing you to use it.

**Comment by ChaosColors** [comment]
I feel like you should save this change for the progression on your expert pack, but i know anything we say won't matter. So good luck on your pack. Maybe throw some clay in there for balancing purposes.

**Comment by dgobi** [comment]
I'm very happy with this change personally! 

As MuteTiefling eloquently put it, nobody is talking about removing flight, step assist, speed, night vision, etc. You can still do all of those things, all early game even. Just in far more interesting ways with a sense of actual progression instead of random luck because you broke 100 leaf blocks.

That said, this is really just a config change to blacklist the trinkets. If it bothers you post-update, you can easily re-enable whichever ones you want for your game by removing them from that list in the configs.

**Comment by stefment** [comment]
So why didnt you just disable it in the config yourself in the first place?

**Comment by NillerMedDild** [comment]
> I feel like you should save this change for the progression on your expert pack, but i know anything we say won't matter. So good luck on your pack. Maybe throw some clay in there for balancing purposes.

What makes you say `i know anything we say won't matter`? 

Given people's strong opinions against this nerf, I'll hold off on blacklisting the trinkets that work until we recommend making a new world in v0.4.0. 
I'll use the time until then to look into other alternatives than direct removals.

**Comment by NillerMedDild** [comment]
> 
> 
> > The thing is, everything that's being disabled can still be obtained through other mods.
> 
> The rock candy speed effect is sort of unique in how it applies forward motion
> And dark egg vexs despawn just fine once their timer runs out similar to how vanilla vexes do.

@TopPercentageRattata We are adding a Paraglider/hangglider mod that also applies forward motion: https://www.curseforge.com/minecraft/mc-mods/paragliders

**Comment by Octr** [comment]
Going to leave my own experience/feedback here just to give some more perspective on the issues.
The reaction of some people on this suggestion is just revealing of the nature of the mod itself, when someone is spoiled with overpowered and unbalanced mechanics they will often have a negative reaction to that thing being changed/taken away.

Not even a day into playing E6 I basically reached godmode due to how the mod is made itself and not even it's interaction with the pack.

The mod is just not balanced in any aspect and it brings those issues with it into the pack, they can not really be resolved or balanced without the mod author making some drastic changes to how the trinkets are earned and how they behave, there is no progression with trinkets themselves, they do not have any real "tiers" except a very minor (and pointless) rarity tier that seems to not actually matter in the long term.

They do not level up, they do not slowly get better, they come as is fully powered with no effort or progression put into them.

The mod is good in "CONCEPT" I love the idea of getting random rewards that improve gameplay from just playing the game and getting different trinkets compared to my friends, the UI and the slot system is also a nice addition and how it is independent from baubles, however EXP is a free resource once you get your first spawner you can unlock all slots without any real effort, which would be better gated by an item you maybe find in dungeons or have to farm for rather than being EXP.
I do not believe this mod was really designed with balance/progression or even modpacks in mind without really just being a "fun" addition to vanilla minecraft.

Fun is great, that's the point in playing but this is a certain type of fun that fundamentally is taking fun away from other areas of the game in the long run.

Just because it is a kitckensink pack doesn't mean there should be no progression, rather it should be progression by choice (not hardlocked) the progression should still exist, If I wanted to start a game as a literal un-killable flying god I could just play in creative.

I think this is a balancing issue with the mod itself, without even considering how it interacts with certain fake players like the Auto Attack Pedestal (which just afk unlocks all trinkets). 

I can not really look at the fact I can't die to anything (lava/any mob / boss etc) because of my trinkets that I put no time or effort into and say "This is fine".

I could see these trinkets being more viable if they had progression to them as well, like you get a trinket from the rng method and then are able to level it up to make it more powerful, through resources or found items.
Reduced fall damage slowly evoling into 0 fall damage with effort and time
Arrow reduced damage slowly evolving into arrow immunity with effort and time
etc

but in it's current state it gives too much for doing too little.

**Comment by dgobi** [comment]
@stefment I did disable the completely broken ones. However I also reported this major balance issue here as you have noted. Not sure what to tell you lol. It's subjective as I stated in my initial post.

In my server several people are completely new to Minecraft and had unlocked the OP trinkets very early (first day playing). They don't see the point of jetpacks and minecarts when they can fly at supersonic speeds by default for free. Your mileage may vary. 

Seems a shame to me.

**Comment by Aelphais** [comment]
Chiming in as well. I've been doing a lot of beta testing dev builds for E6, which has resulted in me created a bunch of worlds and playing through them for a time. In basically all of these worlds, I've gotten at least one broken trinket, possibly more, in the early game several times. As in, before I even build my first non-vanilla machines. I've had complete immunity to fire/lava and poison in most of the worlds I've created very early on, and even creative flight one time on the first day through some truly godly RNG. These things are completely broken and remove any need to use actually well designed features from other mods.

I actually brought this mod up in the discord a few weeks back, asking others opinions on how OP they thought it was, which started a rather lengthy argument. I didn't end up making an issue about it as most people seemed to like it. But seeing this already made, I've decided to weigh in my opinion. If it were up to me, I would remove the mod entirely. Blocking the most problematic ones is a good compromise.

Though if it were possible to relegate the trinkets to things like end city chests instead of randomly getting them, that might be a good idea too. It might even be fun to start seeing some of the low level common trinkets showing up in nether fortresses, then rare trinkets in the end, and the legendary ones just being extremely uncommon rare drops in only the most end game dimensions. Doing so would add a reason to go exploring, excitement and engagement upon finding the trinkets, and a goal to have aside from just building a base. A better play experience, I think, than getting god mode because you rolled a nat 20.

Of course, that brings with it the problem of having a bunch of duplicates. While in a server environment, players could set up a shop or barter for trinkets they don't have (which is even more natural emergent gameplay in a positive direction), single player games would probably wind up with a lot of duplicates. This might be getting into more of a request for the actual mod author than something that goes in E6's issues, and maybe I'll go make a post over there, but I'm going to type it here anyway:

Being able to trade duplicate trinkets for other trinkets would probably be a good use for them. As might being able to use the trinkets to buy additional trinket slots. Something like 4 common trinkets gives you a common wildcard, used to purchase any common trinket or combined with 3 more common wildcards to make a rare trinket. 4 rare wildcards might be able to get a epic trinket or a new trinket slot (disabling the exp expenditure to buy slots). That would give a use for duplicates in a single player world and likely create an economy demand for lots of trinkets in a server environment. But that might all be outside the scope of what the mod author wishes to do.

Anyway, point is, the mod is very overpowered in its current state and _something_ needs to change with it.
